src/main/webapp/app/tasks/local-task.service.ts

```diff
@@ -32,6 +32,23 @@ export class LocalTaskService implements TaskService {
     return of(null);
   }
 
+  togglecomplete(id:string): Observable<void> {
+    const tasks = this.readTasks();
+    tasks.map(task => {
+      if ( task.id === id) {
+      task.completed = !task.completed;
+      this.writeTasks(tasks);
+      }
+    })
+    return of(null);
+  }
+
+  clearcomplete(): Observable<void> {
+    const tasks = this.readTasks();
+    this.writeTasks(tasks.filter(task => task.completed === false));
+    return of(null);
```

Comment:
Do you think it would make sense to return the task here? `of(null)` was used in delete because there simply is no task to return anymore after it is gone.